### DISCUSSION TITLE
api gets stuck on trying to get a repository

### DISCUSSION THREAD

**Discussion by SpieringsAE** [hub4j/github-api#1406] [Q&A] [answered]
Hey everyone, I'm having some issues using the api.

I'm trying to gather update information from a github page but it seems to get stuck at the point i call the getRepository method. The odd thing is that it works sometimes, but most of the time it doesnt.

I'm also having difficulties with the .path filter for the .queryCommits, it either gives me errors or does nothing whatsoever. Its not in the code in this example but the folder I'm trying to filter for is this one: https://github.com/Rick-GO/GOcontroll-Moduline/tree/master/usr/module-firmware.

my code:

`
class CompareVersion extends AsyncTask<Void, Void, Void> {

        @Override

        protected Void doInBackground(Void... voids) {

            Log.d(TAG, "CompareVersion: starting to compare version");

            try {

                gitHub = GitHub.connectAnonymously();

            } catch (IOException e) {

                Log.e(TAG, "CompareVersionn: unable to connect to github", e);

            }

            Log.d(TAG, "CompareVersion: getting repository");

            try {

                repository = gitHub.getRepository("Rick-GO/GOcontroll-Moduline");

            } catch (IOException e) {

                Log.e(TAG, "CompareVersion: unable to get repository", e);

            }

            if (fileURLs != null) {

                fileURLs.clear();

            }

            Log.d(TAG, "CompareVersion: searching for commits");

            for (GHCommit c : repository.queryCommits().since(lastUpdateDate).list()) {

                Log.d(TAG, "CompareVersion: Commit found");

                try {

                    commitDates.add(c.getCommitDate());

                } catch (IOException e) {

                    Log.e(TAG, "CompareVersion: not able to extract commit date from commit", e);

                }

                List<GHCommit.File> updatedFiles = null;

                try {

                    updatedFiles = c.getFiles();

                } catch (IOException e) {

                    Log.e(TAG, "CompareVersion: not able to get files from commit", e);

                }
                assert updatedFiles != null;

                for (GHCommit.File file : updatedFiles) {

                    Log.d(TAG, "CompareVersion: current file: "+ file.getFileName());

                    if (file.getFileName().contains("srec")) {

                        Log.d(TAG, "CompareVersion: srec file found: "+ file.getFileName());

                        fileURLs.add(file.getRawUrl().toString());

                    }

                }

            }

            Log.d(TAG, "CompareVersion: search finished: " + fileURLs.size());

            if (fileURLs.size() > 0) {

                EventBus.getDefault().post(new CompareVersionComplete());

            }

            return null;

        }

    }
`

logcat:

D/MainActivity: CompareVersion: starting to compare version
D/MainActivity: CompareVersion: getting repository
D/NetworkSecurityConfig: No Network Security Config specified, using platform default
W/System: ClassLoader referenced unknown path: system/framework/mediatek-cta.jar
W/System: ClassLoader referenced unknown path: system/framework/mediatek-cta.jar
D/Surface: Surface::disconnect(this=0x6f9af34000,api=1)
D/View: [Warning] assignParent to null: this = com.color.internal.widget.ColorToastLayout{f1dd4d1 V.E...... ......ID 0,0-545,105 #c0204f9 oppo:id/color_toast_layout}
I/InputTransport: Destroy ARC handle: 0x6f9af468e0
I/Choreographer: Skipped 2 frames!  The application may be doing too much work on its main thread.

**Comment by gsmet** [comment, marked as answer]
That might well be your issue: `GitHub.connectAnonymously();`.

You should authenticate with a token. Anonymous access is a lot more chaotic and is rate limited a lot.

**Reply by SpieringsAE** [reply]
So i should generate a token, and then use the connectUsingOAuth(String oauthAccessToken) method? 

Because i dont want my users to have to login to a github account.

Also do you have any input on the path filter for the queryCommits method? It would significantly reduce my update gathering time.

**Reply by gsmet** [reply]
Well, first would be to properly format your code in the description because it's barely readable as it is and I can't figure out what you're doing.

**Reply by SpieringsAE** [reply]
terribly sorry, i didnt realise github had made a mess of it.

the token is working btw thanks a lot for that!

**Reply by gsmet** [reply]
As for `path()`, I don't know how you use it or which errors you got so it's hard to help.

**Reply by SpieringsAE** [reply]
Well that seems to have been resolved aswell apparently, and it now works how i expect it to. thanks for your help, greatly appreciated.

But it's been to far back so i cant dig up up the error anymore

**Reply by gsmet** [reply]
Yeah no worries. They are very defensive against anonymous API usage so avoid it when you can.

**Reply by SpieringsAE** [reply]
I am noticing some weird behaviour with the .since and .from methods in the queryCommitsBuilder.

when i use either of these it will completely ignore it and just search from the start of the repository.

when i add .untill, it just doesnt show anything anymore.

preferably id use .from with the SHA1 since that is easier to update cross platform
`
for (GHCommit c : repository.queryCommits().from("42ffa1e79d4bbc7573c798e4ba9e0aad48050e06").path("/usr/module-firmware").list()) {
`

This is what my for loop becomes when i try to filter with the from method, that sha is a commit from december 30 2021

But when i query it shows me commits as old as like 2020 something.

D/MainActivity: CompareVersion: Commit found: Mon Apr 06 08:02:11 GMT+02:00 2020

for example.

EDIT: it seems that this is a github problem, and these just dont work together. Not even on their website
